tensorflow/java/src/main/java/org/tensorflow/op/core/Constant.java

```diff
@@ -0,0 +1,177 @@
+    try (Tensor value = Tensor.create(shape, data)) {
+      return createWithTensor(scope, value);
+    }
+  }
+
+  /**
+   * Create a {@link DataType#DOUBLE} constant with data from the given buffer.
+   *
+   * <p>Creates a constant with the given shape by copying elements from the buffer (starting from
+   * its current position) into the tensor. For example, if {@code shape = {2,3} } (which represents
+   * a 2x3 matrix) then the buffer must have 6 elements remaining, which will be consumed by this
+   * method.
+   *
+   * @param scope is a scope used to add the underlying operation.
+   * @param shape the tensor shape.
+   * @param data a buffer containing the tensor data.
+   * @throws IllegalArgumentException If the tensor shape is not compatible with the buffer
+   */
+  public static Constant create(Scope scope, long[] shape, DoubleBuffer data) {
+    try (Tensor value = Tensor.create(shape, data)) {
+      return createWithTensor(scope, value);
+    }
+  }
+
+  /**
+   * Create a {@link DataType#INT64} constant with data from the given buffer.
+   *
+   * <p>Creates a constant with the given shape by copying elements from the buffer (starting from
+   * its current position) into the tensor. For example, if {@code shape = {2,3} } (which represents
+   * a 2x3 matrix) then the buffer must have 6 elements remaining, which will be consumed by this
+   * method.
+   *
+   * @param scope is a scope used to add the underlying operation.
+   * @param shape the tensor shape.
+   * @param data a buffer containing the tensor data.
+   * @throws IllegalArgumentException If the tensor shape is not compatible with the buffer
+   */
+  public static Constant create(Scope scope, long[] shape, LongBuffer data) {
+    try (Tensor value = Tensor.create(shape, data)) {
+      return createWithTensor(scope, value);
+    }
+  }
+
+  /**
+   * Create a constant with data from the given buffer.
+   *
+   * <p>Creates a Constant with the provided shape of any type where the constant data has been
+   * encoded into {@code data} as per the specification of the TensorFlow <a
+   * href="https://www.tensorflow.org/code/tensorflow/c/c_api.h">C API</a>.
+   *
+   * @param scope is a scope used to add the underlying operation.
+   * @param dataType the tensor datatype.
+   * @param shape the tensor shape.
+   * @param data a buffer containing the tensor data.
+   * @throws IllegalArgumentException If the tensor datatype or shape is not compatible with the
+   *     buffer
+   */
+  public static Constant create(Scope scope, DataType dataType, long[] shape, ByteBuffer data) {
+    try (Tensor value = Tensor.create(dataType, shape, data)) {
+      return createWithTensor(scope, value);
+    }
+  }
+
+  private static Constant createWithTensor(Scope scope, Tensor value) {
+    return new Constant(
+        scope
+            .graph()
+            .opBuilder(OP, scope.makeOpName(OP))
+            .setAttr(VALUE, value)
```

Comment:
Since this is the one and only place where the string constants are used, might as well inline them